src/prerender.ts

```diff
@@ -201,7 +200,10 @@ export async function prerender(nitro: Nitro) {
 
     const filePath = join(nitro.options.output.publicDir, _route.fileName);
     await writeFile(filePath, Buffer.from(_route.data));
-    nitro._prerenderedRoutes.push(_route);
+    nitro._prerenderedRoutes.push({
```

Comment:
For me, this is the key contribution of this PR and it is a genuine memory leak in nitro. Caching _all_ results of all prerendered routes will of course cause OOM issues.

Reply:
Yes, that alone can be a good fix 👍🏼 (not reordering `_payload.json`).

Reply:
Agreed. Nuxt can hook into prerender hooks hopefully to do the same.